solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/test/java/com/solace/spring/cloud/stream/binder/util/ErrorQueueRepublishCorrelationKeyIT.java

```diff
@@ -211,107 +204,32 @@ public void testHandleErrorAsyncRetry(boolean isDurable, JCSMPSession jcsmpSessi
-		Mockito.verify(errorQueueInfrastructure, Mockito.times(maxDeliveryAttempts)).send(messageContainer, key);
-		assertEquals(maxDeliveryAttempts, key.getErrorQueueDeliveryAttempt());
-		validateNumEnqueuedMessages(vpnName, errorQueue, 0, sempV2Api);
-
-		if (isDurable) {
-			validateNumEnqueuedMessages(vpnName, queue, 1, sempV2Api);
-			Mockito.verify(flowReceiverContainer).acknowledgeRebind(messageContainer, true);
-			assertEquals((Long) 2L, sempV2Api.monitor()
-					.getMsgVpnQueue(vpnName, queue.getName(), null)
-					.getData()
-					.getBindSuccessCount());
-		} else {
-			validateNumEnqueuedMessages(vpnName, queue, 0, sempV2Api);
-		}
-	}
-
-	@ParameterizedTest
-	@ValueSource(booleans = {false, true})
-	public void testHandleErrorRequeueFallbackSkipSync(boolean isDurable, JCSMPSession jcsmpSession, Queue durableQueue,
 													   SempV2Api sempV2Api) throws Exception {
 		String vpnName = (String) jcsmpSession.getProperty(JCSMPProperties.VPN_NAME);
 		Queue queue = isDurable ? durableQueue : jcsmpSession.createTemporaryQueue();
 		FlowReceiverContainer flowReceiverContainer = createFlowReceiverContainer(jcsmpSession, queue);
 
 		producer.send(JCSMPFactory.onlyInstance().createMessage(TextMessage.class), queue);
 		MessageContainer messageContainer = flowReceiverContainer.receive(5000);
-		ErrorQueueRepublishCorrelationKey key = createKey(messageContainer, flowReceiverContainer, isDurable);
+		ErrorQueueRepublishCorrelationKey key = createKey(messageContainer, flowReceiverContainer);
 
 		Mockito.doThrow(new JCSMPException("test")).when(errorQueueInfrastructure).send(messageContainer, key);
 
-		key.handleError(true);
+		key.handleError();
 
 		int maxDeliveryAttempts = Math.toIntExact(errorQueueInfrastructure.getMaxDeliveryAttempts());
 		Mockito.verify(errorQueueInfrastructure, Mockito.times(maxDeliveryAttempts)).send(messageContainer, key);
 		assertEquals(maxDeliveryAttempts, key.getErrorQueueDeliveryAttempt());
 		validateNumEnqueuedMessages(vpnName, errorQueue, 0, sempV2Api);
 
-		if (isDurable) {
-			validateNumEnqueuedMessages(vpnName, queue, 1, sempV2Api);
-			Mockito.verify(retryableTaskService)
-					.submit(new RetryableAckRebindTask(flowReceiverContainer, messageContainer, retryableTaskService));
-			Mockito.verify(flowReceiverContainer).acknowledgeRebind(messageContainer, true);
-			retryAssert(() -> assertEquals((Long) 2L, sempV2Api.monitor()
-					.getMsgVpnQueue(vpnName, queue.getName(), null)
-					.getData()
-					.getBindSuccessCount()));
-		} else {
-			validateNumEnqueuedMessages(vpnName, queue, 0, sempV2Api);
-		}
-	}
+		//Assert: Message re-queued on main queue, for both durable and non-durable queue
+		validateNumEnqueuedMessages(vpnName, queue, 1, sempV2Api);
 
-	@ParameterizedTest
-	@ValueSource(booleans = {false, true})
-	public void testHandleErrorRequeueFallbackFail(boolean isDurable, JCSMPSession jcsmpSession, Queue durableQueue,
```

Comment:
Correct me if I'm wrong, but isn't this test still applicable if the message settlement fails?

Reply:
yes, well identified. 
Restored the test.